glibc_multi: match output ordering of glibc

Artturin opened the pull request:
done with
`sd --string-mode '${stdenv.cc.libc}/lib' '${lib.getLib stdenv.cc.libc}/lib' $(fd -t f)`
Fixes https://github.com/NixOS/nixpkgs/commit/c1fffdfffb43b07ce09b14da0af9a856e1e8566e#commitcomment-74826532
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

vcunat commented:
Out of interest, where does it differ?  Over longer term, perhaps we should fix that for better uniformity?

Quick try on master:
```
nix-repl> toString stdenv.cc.libc              
"/nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210"

nix-repl> toString (lib.getLib stdenv.cc.libc)
"/nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210"
```

Artturin commented:
> Out of curiosity, where does it differ? Over longer term, perhaps we should fix that for better uniformity?
> 
> Quick try on master:
> 
> ```
> nix-repl> toString stdenv.cc.libc              
> "/nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210"
> 
> nix-repl> toString (lib.getLib stdenv.cc.libc)
> "/nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210"
> ```

virtualbox uses `stdenv_32bit`  and glibc_multi has its outputs ordered differently from glibc

the reason glibc has out as its default output and not bin is because of a exception made in [`e065baa` (#14766)](https://github.com/NixOS/nixpkgs/pull/14766/commits/e065baafba77424066e959758681bbcc62c335f7)

```
nix-repl> glibc.outputName
"out"

nix-repl> glibc_multi.outputName
"bin"

nix-repl> glibc.outputs
[ "out" "bin" "dev" "static" "debug" ]

nix-repl> glibc_multi.outputs
[ "bin" "dev" "out" ]
```

Artturin committed:
glibc_multi: match output ordering of glibc

glibc has an exception in that 'out' is the default output instead of 'bin'

it should be matched here for consistency

vcunat commented:
OK.  I knew the glibc exception.  I was just about to post a suggestion to match, but you were faster.  Occurrences of `glibc_multi` seem OK.

vcunat commented:
I wonder if we should drop `getLib` in this case (not add it, I mean), as I'd say that it would be easier to read, and the glibc exception is well-documented.  Though perhaps we might want to extend the docs to any libc; I checked that musl and the *-darwin `stdenv.cc.libc` don't even have any `.bin` output.

vcunat commented:
I checked that `virtualbox` builds again.
